### PR TITLE
Allow blending to work when the output is the foreground

### PR DESCRIPTION
**Pull request by sboukortt**
When copying the background to the output in PrepareBlending, we now leave a hole where the foreground will be, in case it is in fact already there.

### CODE REVIEW

File: lib/jxl/blending.cc [hunk 1]
```diff
@@ -10,6 +10,60 @@
...
+// consistent when called successively with "parallel" rects.
+//
+// +----------------------+
+// |         top          |
+// +------+-------+-------+
+// | left | inner | right |
+// +------+-------+-------+
+// |        bottom        |
+// +----------------------+
+std::vector<Rect> SubtractRect(const Rect& outer, const Rect& inner) {
```
Comment (veluca93) [inline lib/jxl/blending.cc:26]: Maybe I'm just over-optimizing, but how about filling an output array and returning the number?

Comment (sboukortt) [reply lib/jxl/blending.cc:26]: I was thinking of that too but wasn’t sure whether it would be worth it. Let me give it a try.

Comment (sboukortt) [reply lib/jxl/blending.cc]: There we go.

Comment (sboukortt) [reply lib/jxl/blending.cc]: (And just removed the leftover `std::vector<Rect> result;`.)

**Review by veluca93 — APPROVED** [review]

**Pull request merged**